File: LOO 2°Parcial/ComplejosCuaternionesJava/src/ClasesComplejos/Complex.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package ClasesComplejos;

/**
 * 
 * @author Dimitri Medina Novelo
 */
public class Complex extends HiperComplex{
    private double real, imag;
     public Complex(){
     this.real = 0;
    this.imag = 0;
     }
     public Complex(double r, double i){
     this.real = r;
    this.imag = i;
     }
     public void add(Complex n1, Complex n2){
     this.real = n1.getReal() + n2.getReal();
    this.imag = n1.getImag() + n2.getImag();
     }
     public void subs(Complex n1, Complex n2){
     this.real = n1.getReal() - n2.getReal();
    this.imag = n1.getImag() - n2.getImag();
     }
     public void mult(Complex n1, Complex n2){
     this.real = n1.getReal() * n2.getReal() - n1.getImag() * n2.getImag();
    this.imag = n1.getReal() * n2.getImag() + n1.getImag() * n2.getReal();
     }
     public void divi(Complex n1, Complex n2){
     this.real = (n1.getReal() * n2.getReal() + n1.getImag() * n2.getImag()) / (n2.getReal()* n2.getReal() + n2.getImag() * n2.getImag());
    this.imag = (-n1.getReal() * n2.getImag() + n2.getReal() * n1.getImag()) / (n2.getReal() * n2.getReal() + n2.getImag() * n2.getImag());
     }

     public void setReal(double r){
     this.real = r;
     }
     public double getReal(){
     return this.real;
     }
     public void setImag(double i){
     this.imag = i;
     }
     public double getImag() {
     return this.imag;
     }
     
     @Override
     public double module(){
     return Math.sqrt(this.real * this.real + this.imag* this.imag);
     }
     @Override
     public void  conjugate(){
         this.imag *= -1;
     }
     @Override
     public String toString(){
     String str;
    str = (this.real) + (this.imag<0?"":"+") + (this.imag) + " i";
    return str;
     }
     @Override
     public void print(){
         System.out.println((this.real) + (this.imag<0?"":"+") + (this.imag) + " i");
     }
}

File: LOO 2°Parcial/Complex.py

import HiperComplex as HiperC
import math as m

class Complex:
 def __init__(self, r=0, i=0):
  self.__real = r
  self.__imag = i

 @property
 def real(self):
  return self.__real

 @property
 def imag(self):
  return self.__imag

 @real.setter
 def real(self, r):
  self.__real = r

 @imag.setter
 def imag(self, i):
  self.__imag = i

 def module(self):
  return m.sqrt(self.__real*self.__real + self.__imag*self.__imag)

 def conjugate(self):
  return Complex(self.__real, - self.__imag)

 def add(self, n1, n2):
  self.__real = n1.real + n2.real
  self.__imag = n1.imag + n2.imag

 def subs(self, n1, n2):
  self.__real = n1.real - n2.real
  self.__imag = n1.imag - n2.imag

 def mult(self, n1, n2):
  self.__real = n1.real * n2.real - n1.imag * n2.imag
  self.__imag = n1.real * n2.imag + n1.imag * n2.real

 def divi(self, n1, n2):
  self.__real = (n1.real * n2.real + n1.imag * n2.imag) / (n2.real * n2.real + n2.imag * n2.imag)
  self.__imag = (-n1.real * n2.imag + n2.real * n1.imag) / (n2.real * n2.real + n2.imag * n2.imag)

 def toString(self):
     string = str(self.__real)
     if(self.__imag>=0):
        string += "+"
     string += str(self.__imag) + " i"
     return string